cmd/watch.go

```diff
@@ -65,12 +70,16 @@ var watchCmd = &cobra.Command{
 		}
 		watchPath := util.ReadFilePath(u, runtime.GOOS)
 
-		err = component.WatchAndPush(client, componentName, applicationName, watchPath, stdout)
+		err = component.WatchAndPush(client, componentName, applicationName, watchPath, stdout, ignores, delay)
 		checkError(err, "Error while trying to watch %s", watchPath)
 	},
 }
 
 func init() {
+	// ignore git as it can change even if no source file changed
+	// for example some plugins providing git info in PS1 doing that
+	watchCmd.Flags().StringSliceVar(&ignores, "ignores", []string{".*\\.git.*"}, "Files and/or folders to be recursively ignored for watch can be specified as regular expressions with this flag.")
```

Comment:
"ignores" isn't grammatically correct for English. The plural is just "ignore".

Reply:
Ack